packages/expo-task-manager/src/TaskManager.ts

```diff
@@ -2,62 +2,109 @@ import { EventEmitter, UnavailabilityError } from '@unimodules/core';
 
 import ExpoTaskManager from './ExpoTaskManager';
 
-interface TaskError {
+/**
+ * Error object that can be received through TaskManagerTaskBody when the task fails.
+ */
+export interface TaskManagerError {
   code: string | number;
   message: string;
 }
 
-interface TaskBody {
+/**
+ * Represents the object that is passed to the task executor.
+ */
+export interface TaskManagerTaskBody {
+  /**
+   * An object of data passed to the task executor. Its properties depends on the type of the task.
+   */
   data: object;
-  error: TaskError | null;
+
+  /**
+   * Error object if the task failed or `null` otherwise.
+   */
+  error: TaskManagerError | null;
+
+  /**
+   * Additional details containing unique ID of task event and name of the task.
+   */
   executionInfo: {
     eventId: string;
     taskName: string;
   };
 }
 
-export interface RegisteredTask {
+/**
+ * An interface of already registered task.
+ */
+export interface TaskManagerTask {
+  /**
+   * Name under which the task was registered.
+   */
   taskName: string;
+
+  /**
+   * Type of the task which depends on how the task was registered.
+   */
   taskType: string;
+
+  /**
+   * Provides `options` that the task was registered with.
+   */
   options: any;
 }
 
-type Task = (body: TaskBody) => void;
+/**
+ * @deprecated in favor of TaskManagerTask.
+ */
+export interface RegisteredTask extends TaskManagerTask {}
 
-const tasks: Map<string, Task> = new Map<string, Task>();
+/**
+ * Type of task executor – a function that handles the task.
+ */
+export type TaskManagerTaskExecutor = (body: TaskManagerTaskBody) => void;
 
-let isRunningDuringInitialization = true;
+const tasks: Map<string, TaskManagerTaskExecutor> = new Map<string, TaskManagerTaskExecutor>();
 
 function _validateTaskName(taskName) {
   if (!taskName || typeof taskName !== 'string') {
     throw new TypeError('`taskName` must be a non-empty string.');
   }
 }
 
-export function defineTask(taskName: string, task: Task) {
-  if (!isRunningDuringInitialization) {
-    console.error(`TaskManager.defineTask must be called during initialization phase!`);
-    return;
-  }
+/**
+ * Method that you use to define a task – it saves given task executor under given task name
+ * which must be correlated with the task name used when registering the task.
```

Comment:
Your comment is too complicated IMO ... 😢 

Why not to use comment from our docs? 😊
```suggestion
 * Defines task function.
 * It must be called in the global scope of your JavaScript bundle. In particular,
 * it cannot be called in any of React lifecycle methods like `componentDidMount`.
 * This limitation is due to the fact that when the application is launched in the background,
 * we need to spin up your JavaScript app, run your task and then shut down — no views are 
 * mounted in this scenario.
```

Or enrich/enhance it even further 🤔 
```suggestion
 * Defines a task. Provided `taskExecutor` subscribes for the tasks that are registered under `taskName`.
 * It must be called in the global scope of your JavaScript bundle. In particular,
 * it cannot be called in any of React lifecycle methods like `componentDidMount`.
 * This limitation is due to the fact that when the application is launched in the background,
 * we need to spin up your JavaScript app, run your task and then shut down — no views are 
 * mounted in this scenario.
```

Reply:
I didn't want to copy things from the docs - imho our docs should be a place where you can get more details about something and here we should only give some short and basic explanations.